src/components/inputs/time-input/time-input.js

```diff
@@ -53,15 +56,10 @@ const TimeInput = props => {
 
   const onClear = React.useCallback(() => emitChange(''), [emitChange]);
 
-  // so that it doesn't run on initial mount
-  const mounted = React.useRef();
-  React.useEffect(() => {
-    if (!mounted.current) {
-      mounted.current = true;
-    } else {
-      emitChange(TimeInput.toLocaleTime(value, intl.locale));
-    }
-  }, [intl.locale, emitChange, value]);
+  // if locale has changed
+  if (typeof prevLocale !== 'undefined' && prevLocale !== intl.locale) {
```

Comment:
Hmm this looks wrong. As far as I understand it, the dependencies of `useEffect` or `useCallback` are meant exactly for checking if a value changed. So when the body function of those hooks gets executed, it means that one of the deps values changed.

Maybe I'm missing a bit the intention here. Can you clarify when exactly the `emitChange` should be called?

Reply:
Sure, `emitChange` should only be called with `intl.locale` changes.

Here's an gif :)

![like this](https://user-images.githubusercontent.com/2425013/62704379-80bf8900-b9eb-11e9-95e9-01c6f3bb0542.gif)

Reply:
It's to reformat for the currency.

Reply:
another option is removing the undefined check and also do it on mount...

Reply:
AFAIK, either we use a `useEffect` where we do NOT define, `value` in the deps array, or we do it like this. There's no other way.

I would like to merge this and release ui-kit before I go on vacation tonight...

Reply:
Yeah I don't see any other option at the moment.